### PR TITLE
fix: rework timeout with per_request_timeout

### PR DESCRIPTION
**Pull request by tcrasset**
- feat: clarify retry mechanism and custom timeout
- docs: explain timeout and retry mechanism
- refactor: rework timeout with per_request_timeout

### CODE REVIEW

File: doc/source/tutorial.rst [hunk 1]
```diff
@@ -182,6 +185,47 @@ Run avatarization
...
+not finished, it will raise an exception. However, the job is still
+running on the server. You can call ``get_job`` again, as many times as
+needed. If the job is finished, the call finishes too.
+
+You can modify this timeout by passing the ``timeout`` keyword to
+``get_job``.
+
+.. code:: python
+
+   # Will periodically retry for 10 seconds
```
Comment (charlax) [inline doc/source/tutorial.rst:203]: "for" is ambiguous in this context...

File: avatars/models.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
-# API Version 0.3.7
```
Comment (charlax) [inline avatars/models.py:1]: Can you add those files to the generated list?

**Review by charlax — APPROVED** [review]
Thanks! I guess some of those files can be added to the generated list.

File: avatars/models.py (hunk 2 above)
Comment (tcrasset) [reply avatars/models.py:1]: I like to see the changes, especially when we have merge conflicts.

**Pull request merged**